From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
Date: Mon, 29 Jun 2026 20:11:06 +0530
Subject: mei: trace: Make code compatible for older assign_str
 arguments

After changing assign_str to one argument in KV6.10, backporting to
older version need to change parameters with in trace definition.
Go back to old handling.

Signed-off-by: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
---
 drivers/misc/mei/mei-trace.h | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/drivers/misc/mei/mei-trace.h b/drivers/misc/mei/mei-trace.h
index 24fa321..0f9ca96 100644
--- a/drivers/misc/mei/mei-trace.h
+++ b/drivers/misc/mei/mei-trace.h
@@ -26,8 +26,13 @@ TRACE_EVENT(mei_reg_read,
 		__field(u32, val)
 	),
 	TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+		__assign_str(dev, dev_name(dev));
+		__assign_str(reg,reg);
+#else
 		__assign_str(dev);
 		__assign_str(reg);
+#endif
 		__entry->offs = offs;
 		__entry->val = val;
 	),
@@ -45,8 +50,13 @@ TRACE_EVENT(mei_reg_write,
 		__field(u32, val)
 	),
 	TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+		__assign_str(dev, dev_name(dev));
+		__assign_str(reg,reg);
+#else
 		__assign_str(dev);
 		__assign_str(reg);
+#endif
 		__entry->offs = offs;
 		__entry->val = val;
 	),
@@ -64,8 +74,13 @@ TRACE_EVENT(mei_pci_cfg_read,
 		__field(u32, val)
 	),
 	TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+		__assign_str(dev, dev_name(dev));
+		__assign_str(reg,reg);
+#else
 		__assign_str(dev);
 		__assign_str(reg);
+#endif
 		__entry->offs = offs;
 		__entry->val = val;
 	),
-- 
2.43.0
